server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -404,7 +412,11 @@ public String executor() {
             );
             inProgressLatch.await(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
         } catch (Exception e) {
-            logger.error(e.toString());
+            try {
```

Comment:
This code is really awkward, but for some reason there is an `InterruptedException` that is not caught by the `Exception` class inherently, so this is the only way I could get it to work.  I'm open for suggestions if anyone else can come up with a more elegant solution.

Reply:
:o this should not be possible, do you have a way to reproduce that?